core/src/main/resources/error/error-classes.json

```diff
@@ -174,6 +174,17 @@
     "message" : [ "Unsupported data type <typeName>" ],
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_DESERIALIZER" : {
+    "message" : [ "The deserializer is not supported: " ],
+    "subClass" : {
+      "DATA_TYPE_MISMATCH" : {
+        "message" : [ "need a(n) <desiredType> field but got <dataType>." ]
```

Comment:
It would be better to have the same errors here and the next versions (master). I would prefer the approach in the current PR, and hard-code `a(n)` in the error message instead of passing it as a parameter.